reporter/src/main/kotlin/reporters/CycloneDxReporter.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2019 Bosch Software Innovations GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.reporter.reporters
+
+import com.here.ort.model.OrtResult
+import com.here.ort.model.config.CopyrightGarbage
+import com.here.ort.reporter.LicenseTextProvider
+import com.here.ort.reporter.Reporter
+import com.here.ort.reporter.ResolutionProvider
+import com.here.ort.spdx.SpdxLicense
+
+import org.cyclonedx.BomGeneratorFactory
+import org.cyclonedx.CycloneDxSchema
+import org.cyclonedx.model.Bom
+import org.cyclonedx.model.Component
+import org.cyclonedx.model.Hash
+import org.cyclonedx.model.License
+import org.cyclonedx.model.LicenseChoice
+import org.cyclonedx.model.LicenseText
+
+import java.io.OutputStream
+import java.util.UUID
+
+class CycloneDxReporter : Reporter() {
+    override val reporterName = "CycloneDx"
+    override val defaultFilename = "bom.xml"
+
+    private fun mapHash(hash: com.here.ort.model.Hash): Hash? =
+        enumValues<Hash.Algorithm>().find { it.spec == hash.algorithm.toString() }?.let { Hash(it, hash.value) }
+
+    override fun generateReport(
+        outputStream: OutputStream,
+        ortResult: OrtResult,
+        resolutionProvider: ResolutionProvider,
+        licenseTextProvider: LicenseTextProvider,
+        copyrightGarbage: CopyrightGarbage,
+        postProcessingScript: String?
+    ) {
+        val bom = Bom().apply { serialNumber = "urn:uuid:${UUID.randomUUID()}" }
+
+        ortResult.analyzer?.result?.packages?.forEach { (pkg, _) ->
+            val licenseNames = ortResult.getDeclaredLicensesForId(pkg.id) + ortResult.getDetectedLicensesForId(pkg.id)
```

Comment:
For declared licenses you could get them from the package directly: `pkg.declaredLicensesProcessed.allLicenses`

Reply:
Done.